rotkehlchen/chain/optimism/modules/velodrome/decoder.py

```diff
@@ -36,11 +64,224 @@ def __init__(
+                event.notes = f'Swap {event.balance.amount} {crypto_asset.symbol} in {CPT_VELODROME}'  # noqa: E501
+                spend_event = event
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                (event.address in (ROUTER_V2, ROUTER_V1) or event.address in self.pools)
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} as the result of a swap in {CPT_VELODROME}'  # noqa: E501
+                receive_event = event
+            elif (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
+                event.counterparty = CPT_VELODROME
+
+        if spend_event is None or receive_event is None:
+            log.error(
+                'A swap in velodrome pool must have both a spend and a receive event but one or '
+                'both of them are missing for transaction hash: '
+                f'{context.transaction.tx_hash.hex()}. '
+                f'Spend event: {spend_event}, receive event: {receive_event}.',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        maybe_reshuffle_events(
+            ordered_events=[spend_event, receive_event],
+            events_list=context.decoded_events,
+        )
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_velodrome_pool_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] in (REMOVE_LIQUIDITY_EVENT_V2, REMOVE_LIQUIDITY_EVENT_V1):
+            return self._decode_remove_liquidity_events(
+                tx_log=context.tx_log,
+                decoded_events=context.decoded_events,
+            )
+        if context.tx_log.topics[0] == ADD_LIQUIDITY_EVENT:
+            return self._decode_add_liquidity_events(
+                tx_log=context.tx_log,
+                decoded_events=context.decoded_events,
+            )
+        if context.tx_log.topics[0] in (SWAP_V2, SWAP_V1):
+            return self._decode_swap(context=context)
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_velodrome_gauge_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] not in (GAUGE_DEPOSIT_V2, GAUGE_WITHDRAW_V2, CLAIM_REWARDS_V2):
+            return DEFAULT_DECODING_OUTPUT
+
+        user_or_contract_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+        gauge_address = context.tx_log.address
+        raw_amount = hex_or_bytes_to_int(context.tx_log.data)
+        found_event_modifying_balances = False
+        for event in context.decoded_events:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
```

Comment:
here for example you don't do the try/catch. I would just remove it from the other ones and let the error raise naturally